### PR TITLE
[inductor] Clean up AOTInductor runtime interface

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #109673
* #109606
* #109450
* #109498
* __->__ #109436

Summary: Break the use of the AOTInductorParamShape struct in more primitive types; Change the naming of the returned parameters to be consistent with the aoti_torch ones. The actual at::Tensor clean up will come next.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

**Review by chenyang78 — APPROVED** [review]
LGTM. Thanks!

File: torch/csrc/inductor/aot_runtime/interface.h [hunk 1]
```diff
@@ -64,58 +53,61 @@ AOTInductorError AOTInductorModelContainerDelete(
...
     size_t output_idx,
-    const char** output_name_out);
+    const char** ret_output_dtypes);
 
 // Retieves the input shape with the maximum dimension size for each dimension.
 AOTInductorError AOTInductorModelContainerGetMaxInputShape(
     AOTInductorModelContainerHandle container_handle,
     size_t input_idx,
-    AOTInductorParamShape* input_shape);
+    const int64_t** ret_input_sizes,
```
Comment (ezyang) [inline torch/csrc/inductor/aot_runtime/interface.h:103]: This is a preexisting condition, but what's the ownership on the returned `int*`?

**Review by ezyang — APPROVED** [review]

File: torch/csrc/inductor/aot_runtime/interface.h (hunk 1 above)
Comment (desertfire) [reply torch/csrc/inductor/aot_runtime/interface.h:103]: It is borrow, so are other `ret_*`. I can add comments to state that.

**Review by jansel — APPROVED** [review]

**Comment by desertfire** [comment]
Landed with an exported combined PR, https://github.com/pytorch/pytorch/pull/109678

**Pull request closed without merging**